src/openms/include/OpenMS/TRANSFORMATIONS/FEATUREFINDER/MultiplexDeltaMasses.h

```diff
@@ -57,16 +57,48 @@ namespace OpenMS
   class OPENMS_DLLAPI MultiplexDeltaMasses
   {
     public:
+    
+    /**
+     * @brief set of labels associated with a mass shift
+     * 
+     * For example, a set of SILAC labels [Lys8, Lys8, Arg10] would
+     * result in a +26 Da mass shift.
+     */
+    typedef std::multiset<String> LabelSet;
+
+    /**
+     * @brief mass shift with corresponding label set
+     */
+    struct DeltaMass {
+      double mass_shift;
+      LabelSet label_set;
+    };
 
     /**
      * @brief constructor
      */
-    MultiplexDeltaMasses(std::vector<double> ms);
+    MultiplexDeltaMasses();
     
     /**
-     * @brief add a mass shift
+     * @brief constructor
      */
-    void addMassShift(double ms);
+    MultiplexDeltaMasses(std::vector<DeltaMass> dm);
+        
+    /**
+     * @brief add a delta mass
+     */
+    void addDeltaMass(DeltaMass dm);
+    
+    /**
+     * @brief add a delta mass
+     */
+    void addDeltaMass(double m, LabelSet ls);
+    
+    /**
+     * @brief add a delta mass (with a label set consisting of a
+     * single label)
+     */
+    void addDeltaMass(double m, String l);
```

Comment:
this is also not needed and increases the complexity of the class without benefit.If you really need this functionality then please add a constructor to DeltaMass